src/olympia/amo/utils.py

```diff
@@ -352,15 +378,23 @@ def unsubscribe_newsletter(user_profile, basket_id):
     # `fetch_subscribed_newsletters` but since we shouldn't simply error
     # we just fetch it in case something went wrong.
     if not user_profile.basket_token:
-        basket_data = basket.lookup_user(user_profile.email)
-        user_profile.update(basket_token=basket_data['token'])
-
-    try:
-        response = basket.unsubscribe(
-            user_profile.basket_token, user_profile.email, basket_id)
-        return response['status'] == 'ok'
-    except (basket.BasketNetworkException, basket.BasketException):
-        log.exception('basket exception')
+        try:
+            sync_user_with_basket(user_profile)
+        except (basket.BasketNetworkException, basket.BasketException):
+            log.exception('basket exception')
+            return ()
```

Comment:
Why `()` ?

Reply:
Could very well be an empty list, I don't think I had a good reason to settle on an empty tuple instead. Let me switch that to a list since if something get's returned we do return a list.

Reply:
accidentally, I'll return pass through to the `return False` at the end.

Reply:
Fixed in 339c9bab2c